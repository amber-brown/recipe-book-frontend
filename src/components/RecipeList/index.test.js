import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import { RecipeList } from "./index.js";

const recipes = [
  {
    title: "Key Lime Pie",
    time: { preparation: 30 },
    serves: 200,
    image: "/image.jpeg"
  },
  {
    title: "Fish and Chips",
    time: { preparation: 50 },
    serves: 60,
    image: "/image.jpeg"
  }
];

describe("RecipeList", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <Router>
        <RecipeList getAllRecipes={() => {}} recipes={recipes} />
      </Router>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Router>
          <RecipeList getAllRecipes={() => {}} recipes={recipes} />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
