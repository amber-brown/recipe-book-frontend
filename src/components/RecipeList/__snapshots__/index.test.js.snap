// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeList renders correctly 1`] = `
<div
  className="recipe-list"
>
  <h2>
    Recipes
  </h2>
  <div
    className="list"
  >
    <div
      className="recipe-card"
      style={
        Object {
          "backgroundImage": "url(/image.jpeg)",
        }
      }
    >
      <div
        className="information"
      >
        <h2>
          Key Lime Pie
        </h2>
        <div
          className="sub-information"
        >
          <p>
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-stopwatch fa-w-14 icon"
              data-icon="stopwatch"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M432 304c0 114.9-93.1 208-208 208S16 418.9 16 304c0-104 76.3-190.2 176-205.5V64h-28c-6.6 0-12-5.4-12-12V12c0-6.6 5.4-12 12-12h120c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-28v34.5c37.5 5.8 71.7 21.6 99.7 44.6l27.5-27.5c4.7-4.7 12.3-4.7 17 0l28.3 28.3c4.7 4.7 4.7 12.3 0 17l-29.4 29.4-.6.6C419.7 223.3 432 262.2 432 304zm-176 36V188.5c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12V340c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
            30
             minutes
          </p>
          <p>
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-user-friends fa-w-20 icon"
              data-icon="user-friends"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 640 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M192 256c61.9 0 112-50.1 112-112S253.9 32 192 32 80 82.1 80 144s50.1 112 112 112zm76.8 32h-8.3c-20.8 10-43.9 16-68.5 16s-47.6-6-68.5-16h-8.3C51.6 288 0 339.6 0 403.2V432c0 26.5 21.5 48 48 48h288c26.5 0 48-21.5 48-48v-28.8c0-63.6-51.6-115.2-115.2-115.2zM480 256c53 0 96-43 96-96s-43-96-96-96-96 43-96 96 43 96 96 96zm48 32h-3.8c-13.9 4.8-28.6 8-44.2 8s-30.3-3.2-44.2-8H432c-20.4 0-39.2 5.9-55.7 15.4 24.4 26.3 39.7 61.2 39.7 99.8v38.4c0 2.2-.5 4.3-.6 6.4H592c26.5 0 48-21.5 48-48 0-61.9-50.1-112-112-112z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
            Serves: 
            200
          </p>
        </div>
      </div>
    </div>
    <div
      className="recipe-card"
      style={
        Object {
          "backgroundImage": "url(/image.jpeg)",
        }
      }
    >
      <div
        className="information"
      >
        <h2>
          Fish and Chips
        </h2>
        <div
          className="sub-information"
        >
          <p>
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-stopwatch fa-w-14 icon"
              data-icon="stopwatch"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M432 304c0 114.9-93.1 208-208 208S16 418.9 16 304c0-104 76.3-190.2 176-205.5V64h-28c-6.6 0-12-5.4-12-12V12c0-6.6 5.4-12 12-12h120c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-28v34.5c37.5 5.8 71.7 21.6 99.7 44.6l27.5-27.5c4.7-4.7 12.3-4.7 17 0l28.3 28.3c4.7 4.7 4.7 12.3 0 17l-29.4 29.4-.6.6C419.7 223.3 432 262.2 432 304zm-176 36V188.5c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12V340c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
            50
             minutes
          </p>
          <p>
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-user-friends fa-w-20 icon"
              data-icon="user-friends"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 640 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M192 256c61.9 0 112-50.1 112-112S253.9 32 192 32 80 82.1 80 144s50.1 112 112 112zm76.8 32h-8.3c-20.8 10-43.9 16-68.5 16s-47.6-6-68.5-16h-8.3C51.6 288 0 339.6 0 403.2V432c0 26.5 21.5 48 48 48h288c26.5 0 48-21.5 48-48v-28.8c0-63.6-51.6-115.2-115.2-115.2zM480 256c53 0 96-43 96-96s-43-96-96-96-96 43-96 96 43 96 96 96zm48 32h-3.8c-13.9 4.8-28.6 8-44.2 8s-30.3-3.2-44.2-8H432c-20.4 0-39.2 5.9-55.7 15.4 24.4 26.3 39.7 61.2 39.7 99.8v38.4c0 2.2-.5 4.3-.6 6.4H592c26.5 0 48-21.5 48-48 0-61.9-50.1-112-112-112z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
            Serves: 
            60
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
